framework/test/gtest/LDAPExprTest.cpp

```diff
@@ -242,3 +243,50 @@ TEST(LDAPExprTest, ParseExceptions)
   // Testing '\\' case in LDAPExpr::ParseState::getAttributeValue()
   ASSERT_EQ(LDAPFilter("(name=ab\\a)"), LDAPFilter("(name=aba)"));
 }
+
+TEST(LDAPExprTest, BitWiseOperatorOr)
+{
+  LDAPPropExpr checkedExpr((LDAPProp("key1") == "value1") || (LDAPProp("key2") == "value2"));
+
+  LDAPPropExpr expr(LDAPProp("key1") == "value1");
+
+  expr |= LDAPProp("key2") == "value2";
+  ASSERT_EQ(expr.operator std::string(), checkedExpr.operator std::string());
```

Comment:
Not hung up about it but the implementation of `LDAPPropExpr::operator==` will make this comparison cleaner i.e. `ASSERT_EQ(expr, checkedExpr)`. (may benefit clients too)